docs/Developers/Linter_Bears_Advanced.rst

```diff
@@ -41,6 +40,28 @@ The path of the temporary configuration file can be accessed inside
         def create_arguments(filename, file, config_file):
             return "--use-config", config_file
 
+If you return ``None``, no configuration file is generated. A common case
+where to explicitly return ``None`` is when the user wants to use his own
+tool-specific config. If a config file is supplied by user that means we
+don't have to generate a config file. Generating config file consumes
+resources. Avoiding generating config file increases the performance.
+
+::
+
+    @linter(executable='...')
+    class MyBear:
+        def generate_config(filename, file, settingA: bool):
+            if settingA:
+                return 'A=True'
+            else:
+                return None
+
+    def create_arguments(filename, file, config_file, settingA: bool):
+        args = ['--lint', '--mode', 'standard']
+        if settingA:
+            return '--use-config', config_file
```

Comment:
you should incorporate `filename` into the arguments. Also you don't need `--mode` and `standard`, but you always have to append `--use-config` (depending on whether a user config was supplied, put there `user_config` or `config_file`)
-->
```
return filename, '--use-config', user_config if user_config else config_file
```

Reply:
And then you don't need to provide `settingA` / `setting_a` in the signature altogether :)

Reply:
only remove `settingA` inside `create_arguments`, you don't need it there :+1: But we need it for `generate_config?` :)

> Also now I'll have to add user_config as input argument. Right?

yes :+1: